multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -682,16 +683,33 @@ def fit(
             return self
 
         if self._problem_type == OBJECT_DETECTION:
-            self._detection_anno_train = train_data
-            train_data = from_coco_or_voc(train_data, "train")
-            if tuning_data is not None:
-                self.detection_anno_train = tuning_data
-                tuning_data = from_coco_or_voc(tuning_data, "val")
-                if max_num_tuning_data is not None:
-                    if len(tuning_data) > max_num_tuning_data:
-                        tuning_data = tuning_data.sample(
-                            n=max_num_tuning_data, replace=False, random_state=seed
-                        ).reset_index(drop=True)
+            if isinstance(train_data, str):
+                self._detection_anno_train = train_data
+                train_data = from_coco_or_voc(train_data, "train")  # TODO: Refactor to use convert_data_to_df
+                if tuning_data is not None:
+                    self.detection_anno_train = tuning_data
+                    tuning_data = from_coco_or_voc(tuning_data, "val")  # TODO: Refactor to use convert_data_to_df
+                    if max_num_tuning_data is not None:
+                        if len(tuning_data) > max_num_tuning_data:
+                            tuning_data = tuning_data.sample(
+                                n=max_num_tuning_data, replace=False, random_state=seed
+                            ).reset_index(drop=True)
+            elif isinstance(train_data, pd.DataFrame):
+                self._detection_anno_train = None
+                # sanity check dataframe columns
+                train_data = autogluon.multimodal.utils.object_detection.convert_data_to_df(train_data)
+                if tuning_data is not None:
+                    self.detection_anno_train = tuning_data
+                    tuning_data = autogluon.multimodal.utils.object_detection.convert_data_to_df(tuning_data)
+                    if max_num_tuning_data is not None:
+                        if len(tuning_data) > max_num_tuning_data:
+                            tuning_data = tuning_data.sample(
+                                n=max_num_tuning_data, replace=False, random_state=seed
+                            ).reset_index(drop=True)
+            else:
+                raise TypeError(
+                    f"Expected train_data to have type str or pd.DataFrame, but got type: {type(train_data)}"
+                )
```

Comment:
Can we wrap this part into a util function of `object_detection.py`? Let's simplify `predictor.py`.

Reply:
yup